src/XMonad/Main.hs

```diff
@@ -368,7 +368,7 @@ handle e@(CrossingEvent {ev_window = w, ev_event_type = t})
         dpy <- asks display
         root <- asks theRoot
         (_, _, w', _, _, _, _, _) <- io $ queryPointer dpy root
-        when (w == w') (focus w)
+        when (w' == 0 || w == w') (focus w)
```

Comment:
Is `w' == 0` testing to see if `w'` is the root window?  If so please just compare to the `root` binding and I'll merge.

Reply:
I also wanted to eliminate the magical `0` there, but my knowledge about xlib is limited:
`w' == root` is not the same as `w' == 0`, because turns out `root` is not `0` in this case.
What I do know is by debugging I found that when a mouse is moved to another workspace but without moving into any window, `w'` becomes `0` - I've searched some docs but still don't know what that `0` means.

I've also tried changing the binding line to `(flg, _, w', x', y', wx', wy', _) <- io $ queryPointer dpy root`, but `flg` is always `True`, none of `x'`, `y'`, `wx'` or `wy'` seems to be useful either.

Reply:
From @geekosaur's comment it sounds like you should change `0` to `None`.

Reply:
Not that simple, I'm afraid; I was quoting the C API, and the Haskell wrapper can't use `None` that way. The C `None` is *not* reflected through to Haskell; unless you want to add something to the X11 package, `0` is correct.

Reply:
Okay, @Javran it sounds like the correct solution is to just add a comment describing what `0` is.  After that I'll merge the PR.